lib/matplotlib/backends/backend_pgf.py

```diff
@@ -168,26 +168,28 @@ def _metadata_to_str(key, value):
 
 def make_pdf_to_png_converter():
     """Return a function that converts a pdf file to a png file."""
-    if shutil.which("pdftocairo"):
-        def cairo_convert(pdffile, pngfile, dpi):
-            cmd = ["pdftocairo", "-singlefile", "-png", "-r", "%d" % dpi,
-                   pdffile, os.path.splitext(pngfile)[0]]
-            subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-        return cairo_convert
+    try:
+        mpl._get_executable_info("pdftocairo")
+    except mpl.ExecutableNotFoundError:
+        pass
+    else:
+        return lambda pdffile, pngfile, dpi: subprocess.check_output(
+            ["pdftocairo", "-singlefile", "-transp", "-png", "-r", "%d" % dpi,
+             pdffile, os.path.splitext(pngfile)[0]],
+            stderr=subprocess.STDOUT)
     try:
         gs_info = mpl._get_executable_info("gs")
     except mpl.ExecutableNotFoundError:
         pass
     else:
-        def gs_convert(pdffile, pngfile, dpi):
-            cmd = [gs_info.executable,
-                   '-dQUIET', '-dSAFER', '-dBATCH', '-dNOPAUSE', '-dNOPROMPT',
-                   '-dUseCIEColor', '-dTextAlphaBits=4',
-                   '-dGraphicsAlphaBits=4', '-dDOINTERPOLATE',
-                   '-sDEVICE=png16m', '-sOutputFile=%s' % pngfile,
-                   '-r%d' % dpi, pdffile]
-            subprocess.check_output(cmd, stderr=subprocess.STDOUT)
-        return gs_convert
+        return lambda pdffile, pngfile, dpi: subprocess.check_output(
```

Comment:
why rewrite this as a lambda?

Reply:
For symmetry with the case above; as to why both are lambdas: to avoid running afoul of pep8's request of blank lines around function definitions (even if we don't explicitly check that here) without introducing a bunch of blank lines.

When immediately returning a function, I also find `return lambda ...` easier to read (when the lambda body is indeed a single expression), as that puts the intent ("this is what I'm returning") earlier, but YMMV...

Reply:
No opinion on functions vs lambdas. 😎